hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NodeEndpoint.java

```diff
@@ -171,4 +184,144 @@ private DatanodeStorageReport getStorageReport(DatanodeDetails datanode) {
+    Preconditions.checkArgument(!uuids.isEmpty(), "Datanode list argument should not be empty");
+    try {
+      for (String uuid : uuids) {
+        DatanodeDetails nodeByUuid = nodeManager.getNodeByUuid(uuid);
+        try {
+          if (preChecksSuccess(nodeByUuid, failedNodeErrorResponseMap)) {
+            removedDatanodes.add(DatanodeMetadata.newBuilder()
+                .withHostname(nodeManager.getHostName(nodeByUuid))
+                .withUUid(uuid)
+                .withState(nodeManager.getNodeStatus(nodeByUuid).getHealth())
+                .build());
+            nodeManager.removeNode(nodeByUuid);
+          } else {
+            failedDatanodes.add(DatanodeMetadata.newBuilder()
+                .withHostname(nodeManager.getHostName(nodeByUuid))
+                .withUUid(uuid)
+                .withOperationalState(nodeByUuid.getPersistedOpState())
+                .withState(nodeManager.getNodeStatus(nodeByUuid).getHealth())
+                .build());
+          }
+        } catch (NodeNotFoundException nnfe) {
+          LOG.error("Selected node {} not found : {} ", uuid, nnfe);
+          notFoundDatanodes.add(DatanodeMetadata.newBuilder()
+                  .withHostname("")
+                  .withState(NodeState.DEAD)
+              .withUUid(uuid).build());
+        }
+      }
+    } catch (Exception exp) {
+      LOG.error("Unexpected Error while removing datanodes : {} ", exp);
+      throw new WebApplicationException(exp, Response.Status.INTERNAL_SERVER_ERROR);
+    }
+
+    RemoveDataNodesResponseWrapper removeDataNodesResponseWrapper = new RemoveDataNodesResponseWrapper();
+
+    if (!failedDatanodes.isEmpty()) {
+      DatanodesResponse failedNodesResp =
+          new DatanodesResponse(failedDatanodes.size(), Collections.emptyList());
+      failedNodesResp.setFailedNodeErrorResponseMap(failedNodeErrorResponseMap);
+      removeDataNodesResponseWrapper.getDatanodesResponseMap().put("failedDatanodes", failedNodesResp);
+    }
+
+    if (!notFoundDatanodes.isEmpty()) {
+      DatanodesResponse notFoundNodesResp =
+          new DatanodesResponse(notFoundDatanodes.size(), notFoundDatanodes);
+      removeDataNodesResponseWrapper.getDatanodesResponseMap().put("notFoundDatanodes", notFoundNodesResp);
+    }
+
+    if (!removedDatanodes.isEmpty()) {
+      DatanodesResponse removedNodesResp =
+          new DatanodesResponse(removedDatanodes.size(), removedDatanodes);
+      removeDataNodesResponseWrapper.getDatanodesResponseMap().put("removedDatanodes", removedNodesResp);
+    }
+    return Response.ok(removeDataNodesResponseWrapper).build();
+  }
+
+  private boolean preChecksSuccess(DatanodeDetails nodeByUuid, Map<String, String> failedNodeErrorResponseMap)
+      throws NodeNotFoundException {
+    if (null == nodeByUuid) {
+      throw new NodeNotFoundException("Node  not found !!!");
+    }
+    NodeStatus nodeStatus = null;
+    AtomicBoolean isContainerOrPipeLineOpen = new AtomicBoolean(false);
```

Comment:
Can we Replace AtomicBoolean with a simple boolean if threading isn't a concern.?

Reply:
Apologies, I did not see the     writeLock().lock(); in the removeNode() method 
So I guess we can ignore this comment.